### ISSUE TITLE
Doc: Missing fields in Client return object

### ISSUE THREAD

**Issue by twoseat** [cloudfoundry/uaa#412]
The apparently mandatory field `action`, and the optional fields `access_token_validity` and `refresh_token_validity`, are not documented in http://docs.cloudfoundry.com/uaa/#clients. (Found when performing a `client list` against CF)

**Comment by Priyata25** [comment]
@twoseat, we have updated the docs for the access_token_validity and refresh_token_validity in the client documentation and you will see those changes soon.

We are not sure about the `action` field that you report seeing. There is an `action` field in the "mixed actions" endpoint (http://docs.cloudfoundry.org/api/uaa/#mixed-actions) which specifies how to treat each provided client JSON. If you are seeing an `action` field returned in a client list, it is possible the client was created via a `POST` or altered with a `PUT` with an `action` field erroneously included. The UAA will capture unknown fields in a client JSON and return them.

Could you clarify under what conditions you see this `action` field?

**Comment by twoseat** [comment]
@Priyata25 Thanks for the response! I'm running against pcfdev (currently 0.20). Starting with a clean system (cf dev destroy, cf dev start) if I issue the command:

`uaac clients`

I get the following output. Note the existence of `action` fields.

```
  admin
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials
    autoapprove: 
    action: none
    authorities: clients.read password.write clients.secret clients.write uaa.admin scim.write scim.read
    lastmodified: 1474972090000
  apps_manager_js
    scope: cloud_controller.read cloud_controller.write openid password.write cloud_controller.admin scim.read scim.write doppler.firehose uaa.user routing.router_groups.read
    resource_ids: none
    authorized_grant_types: implicit password
    autoapprove: true
    action: none
    authorities: uaa.none
    lastmodified: 1474972090000
  cc_routing
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials
    autoapprove: 
    action: none
    authorities: routing.router_groups.read
    lastmodified: 1474972090000
  cc-service-dashboards
    scope: openid cloud_controller_service_permissions.read
    resource_ids: none
    authorized_grant_types: client_credentials
    autoapprove: 
    action: none
    authorities: clients.read clients.admin clients.write
    lastmodified: 1474972090000
  cf
    scope: cloud_controller.read cloud_controller.write openid password.write cloud_controller.admin scim.read scim.write doppler.firehose uaa.user routing.router_groups.read routing.router_groups.write
    resource_ids: none
    authorized_grant_types: implicit password refresh_token
    autoapprove: 
    access_token_validity: 600
    refresh_token_validity: 2592000
    action: none
    authorities: uaa.none
    lastmodified: 1474972090000
  cloud_controller_username_lookup
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials
    autoapprove: 
    action: none
    authorities: scim.userids
    lastmodified: 1474972090000
  doppler
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: refresh_token authorization_code
    autoapprove: 
    action: none
    authorities: uaa.resource
    lastmodified: 1474972090000
  gorouter
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials refresh_token
    autoapprove: 
    action: none
    authorities: routing.routes.read
    lastmodified: 1474972091000
  login
    scope: openid oauth.approvals
    resource_ids: none
    authorized_grant_types: authorization_code client_credentials refresh_token
    redirect_uri: https://login.local.pcfdev.io
    autoapprove: true
    action: none
    authorities: clients.read emails.write scim.userids password.write notifications.write oauth.login scim.write critical_notifications.write
    lastmodified: 1474972091000
  notifications
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials
    autoapprove: 
    action: none
    authorities: scim.read cloud_controller.admin
    lastmodified: 1474972091000
  p-mysql
    scope: cloud_controller_service_permissions.read openid
    resource_ids: none
    authorized_grant_types: authorization_code refresh_token
    redirect_uri: http://mysql-broker.local.pcfdev.io/
    autoapprove: 
    action: none
    authorities: uaa.none
    lastmodified: 1474972361000
  ssh-proxy
    scope: openid cloud_controller.read cloud_controller.write
    resource_ids: none
    authorized_grant_types: authorization_code refresh_token
    redirect_uri: /login
    autoapprove: true
    action: none
    authorities: uaa.none
    lastmodified: 1474972091000
  tcp_emitter
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials refresh_token
    autoapprove: 
    action: none
    authorities: routing.routes.write routing.router_groups.read routing.routes.read
    lastmodified: 1474972090000
  tcp_router
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials refresh_token
    autoapprove: 
    action: none
    authorities: routing.routes.read
    lastmodified: 1474972090000
```

I also checked on one of our non-local test system and got the following (partial) output, which makes me think it's not just something non-standard that pcfdev is doing:
```
  admin
    scope: uaa.none
    resource_ids: none
    authorized_grant_types: client_credentials
    autoapprove: 
    action: none
    authorities: clients.read password.write clients.secret clients.write uaa.admin scim.write scim.read
    lastmodified: 1472848771969
  apps_metrics
    scope: cloud_controller.admin cloud_controller.read metrics.read
    resource_ids: none
    authorized_grant_types: authorization_code refresh_token
    redirect_uri: https://apm.magenta.springapps.io https://apm.magenta.springapps.io/ https://apm.magenta.springapps.io/* https://metrics.magenta.springapps.io https://metrics.magenta.springapps.io/
        https://metrics.magenta.springapps.io/*
    autoapprove: 
    access_token_validity: 900
    refresh_token_validity: 2628000
    action: none
    authorities: uaa.none
    lastmodified: 1472848772836
```

**Comment by Priyata25** [comment]
@twoseat Hello, thank you for reporting this issue.

This problem is fixed in the UAA codebase as of release 3.5.0. The problem should disappear from PCF with our next release. That field should not appear, and be assured it has no effect on the client.

I am marking this issue closed. Thank you, and please continue to report gaps in our documentation as you find them.

**Comment by twoseat** [comment]
@Priyata25 So the solution is that `action` no longer appears in the return object (unless someone hacked it with an errant POST)?
